src/hidpp20.c

```diff
@@ -2458,6 +2458,17 @@ hidpp20_onboard_profiles_parse(struct hidpp20_device *device,
 							    profiles,
 							    i,
 							    check_crc);
+
+		/* on fail to read the user profile fallback to the firmware profile */
+		if(rc && profiles->profiles[i].address == i + 1)
+		{
+			profiles->profiles[i].address += 0x0100;
+			rc = hidpp20_onboard_profiles_parse_profile(device,
+							    profiles,
+							    i,
+							    check_crc);
+		}
+
 		if (rc == -EAGAIN)
 			crc_valid = false;
```

Comment:
This makes me think we should rethink the calls instead of just adding a band aid.

Reply:
Yes, we should also probably replace `hidpp20_onboard_profiles_initialize`'s fallback mechanic for this one.